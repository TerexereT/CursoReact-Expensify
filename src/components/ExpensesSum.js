import React from 'react'
import numeral from 'numeral'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const aux = expenseCount === 1 ? 'expensa' : 'expensas'
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viendo <span>{expenseCount}</span> {aux} de un total de <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button-layout" to="/create">Agregar expensa</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect (mapStateToProps)(ExpensesSummary)
