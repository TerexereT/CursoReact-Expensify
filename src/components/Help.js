import React from 'react'

const HelpExpenseDBPage = () => (
    <div>
        esto viene de Help Dashboard
    </div>
)

export default HelpExpenseDBPage