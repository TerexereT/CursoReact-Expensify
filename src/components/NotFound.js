import React from 'react'
import {Link} from 'react-router-dom'

const NotFoundDBPage = () => (
    <div>
        error 404! Page Not Found
        - <Link to="/">Home</Link>
    </div>
)

export default NotFoundDBPage