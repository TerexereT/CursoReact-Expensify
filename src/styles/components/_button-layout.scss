.button-layout {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-l;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}


.button--link {
    background: none;
    
}

.button-layout--secondary {
    @extend .button-layout;
    background: rgb(201, 0, 0);
}