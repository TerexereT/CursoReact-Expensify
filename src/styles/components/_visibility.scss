.show-for-mobile {
    @media (min-width: $desktopBp) {
        display: none;
    }
}

.show-for-desktop {
    @media (max-width: $desktopBp - .01rem){
        display: none;
    }
}