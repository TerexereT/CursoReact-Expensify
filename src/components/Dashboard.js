import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSum from './ExpensesSum'

const ExpenseDBPage = () => (
    <div>
        <ExpensesSum />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDBPage