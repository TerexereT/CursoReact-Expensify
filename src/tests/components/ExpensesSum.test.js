import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSum} from '../../components/ExpensesSum'

test('ver si muestra para 1 expense', () =>{
    const wrapper = shallow(
        <ExpensesSum 
        expenseCount={1} ExpensesTotal={1306}/>
    )
    expect(wrapper).toMatchSnapshot()
})

test('ver si muestra para 2 o mas expenses', () =>{
    const wrapper = shallow(
        <ExpensesSum 
        expenseCount={23} ExpensesTotal={123565345}/>
    )
    expect(wrapper).toMatchSnapshot()
})