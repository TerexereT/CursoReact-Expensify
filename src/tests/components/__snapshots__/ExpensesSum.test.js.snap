// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ver si muestra para 1 expense 1`] = `
<div>
  <h1>
    Viendo 
    1
     
    expense
     de un total de 
    $0.00
  </h1>
</div>
`;

exports[`ver si muestra para 2 o mas expenses 1`] = `
<div>
  <h1>
    Viendo 
    23
     
    expenses
     de un total de 
    $0.00
  </h1>
</div>
`;
